sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -735,70 +735,98 @@ class CodegenContext {
   }
 
   /**
-   * Generates code creating a [[UnsafeArrayData]].
+   * Generates code creating a [[UnsafeArrayData]] or [[GenericArrayData]] based on
+   * given parameters.
    *
    * @param arrayName name of the array to create
+   * @param elementType data type of the elements in source array
    * @param numElements code representing the number of elements the array should contain
-   * @param elementType data type of the elements in the array
    * @param additionalErrorMessage string to include in the error message
+   * @param elementSize optional value which shows the size of an element of the allocated
+   *                    [[UnsafeArrayData]] or [[GenericArrayData]]
+   *
+   * @return code representing the allocation of [[ArrayData]]
+   *         code representing a setter of an assignment for the generated array
    */
-  def createUnsafeArray(
+  def createArrayData(
```

Comment:
I think that it would be good to have a utility method to generate this call. WDYT?
On the other hand, I agree that we have a method `setArrayElementFunc` or `setArrayElement`. Thus, we can split this method into two methods that have only one return value.

Reply:
I'm still not sure if it would be good to have this. I think the method is small enough to inline. cc @cloud-fan

Reply:
I think it would be good to have this since this utility method will be used at other places where `new UnsafeArrayData` or `new GenericArrayData` is generated as Java code.